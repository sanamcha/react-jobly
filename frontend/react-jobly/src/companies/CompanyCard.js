import React from "react";
import { Link } from "react-router-dom";



function CompanyCard({ name, description, logoUrl, handle }) {
    console.log("CompanyCard", logoUrl);
return (
    <Link className="CompanyCard" to={`/companies/${handle}`}>
        <div>
            <h6>
                {name}
                {logoUrl && <img src={logoUrl}
                alt={name} />}
            </h6>
            <p>{description}</p>
        </div>
    </Link>
);

}

export default CompanyCard;