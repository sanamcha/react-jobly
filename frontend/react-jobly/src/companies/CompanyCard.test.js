import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";


it("matches snapshot with logo", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <CompanyCard
            handle="Springboard"
            name="Springboard School"
            description="Become an developer in 36 weeks."
            logo_url="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
        />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot without logo", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <CompanyCard
            handle="nologo"
            name="No-Logo School"
            description="Become a developer in 100 weeks."
        />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
