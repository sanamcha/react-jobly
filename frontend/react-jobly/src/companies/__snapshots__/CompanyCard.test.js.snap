// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard card"
    href="/companies/Springboard"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      >
        Springboard School
      </h6>
      <p>
        Become an developer in 36 weeks.
      </p>
    </div>
  </a>
</DocumentFragment>
`;

exports[`matches snapshot without logo 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard card"
    href="/companies/nologo"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      >
        No-Logo School
      </h6>
      <p>
        Become a developer in 100 weeks.
      </p>
    </div>
  </a>
</DocumentFragment>
`;
